eWonConnector-gateway/src/main/java/org/imdc/ewon/SyncManager.java

```diff
@@ -328,7 +331,8 @@ protected void updateTagValues(EwonData data) {
                         provider.registerWriteHandler(p, new WriteHandler() {
                             public QualityCode write(TagPath tagPath, Object o) {
                                 try {
-                                    comm.writeTag(tagPath.getParentPath().getItemName(), tagPath.getItemName(), o.toString());
+                                    String tagName = replaceUnderscore ? unSanitizeName(tagPath.getItemName()) : tagPath.getItemName();
```

Comment:
Couldn't replaceUnderscore here just be a direct call to settings.isReplaceUnderscore()? That'd eliminate the need for the replaceUnderscore variable it looks like.

Reply:
Not really, settings.isReplaceUnderscore() will read the currently configured value from the ignitions settings page which is expensive as it is reading from the UI and should not be done "mid program".  For this module the settings should only be read on module startup to provide consistency for the settings.  If that setting were changed mid operation by a user it would break the application as the tag names would be not be written to the flexy correctly.  

The issue we are solving here is that 
tag.name is a valid tag on the flexy but not in ignition so we convert it to tag_name in ignition.  Then when we go to write it back to the flexy via talk2m we have to know that _ means . and we have to write tag_name as tag.name.  If the user changed that mid operation we would try and write back tag_name which doesn't exist.

Reply:
Thanks for the clarification. That makes more sense now.